components/locid/src/parser/mod.rs

```diff
@@ -7,9 +7,79 @@ mod langid;
 mod locale;
 
 pub use errors::ParserError;
-pub use langid::{parse_language_identifier, parse_language_identifier_from_iter, ParserMode};
+pub use langid::{
+    parse_language_identifier, parse_language_identifier_from_iter,
+    parse_language_identifier_without_variants, ParserMode,
+};
 pub use locale::parse_locale;
 
-pub fn get_subtag_iterator(t: &[u8]) -> impl Iterator<Item = &[u8]> {
-    t.split(|c| *c == b'-' || *c == b'_')
+pub const fn get_subtag_iterator(t: &[u8]) -> SubtagIterator {
+    let mut l_cursor = 0;
+    while l_cursor < t.len() && (t[l_cursor] == b'_' || t[l_cursor] == b'-') {
+        l_cursor += 1;
+    }
+    let mut r_cursor = l_cursor;
+    while r_cursor < t.len() && t[r_cursor] != b'_' && t[r_cursor] != b'-' {
+        r_cursor += 1;
+    }
+    SubtagIterator {
+        t,
+        l_cursor,
+        r_cursor,
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
+pub struct SubtagIterator<'a> {
+    t: &'a [u8],
+    l_cursor: usize,
+    r_cursor: usize,
```

Comment:
`suggestion`: `t` is a very cryptic name for a field that is carried around. Can you call it `slice` or `source`? For `l_cursor` and `r_cursor` - consider `start`/`end`. or `start_idx` and `end_idx`.